xdem/coreg.py

```diff
@@ -190,13 +190,14 @@ def residuals(parameters: tuple[float, float, float], y_values: np.ndarray, x_va
 
     # Estimate the a, b, and c parameters with least square minimisation
     np.random.seed(seed=42)
-    plsq = scipy.optimize.leastsq(func=residuals, x0=initial_guess, args=(y_medians, slice_bounds), full_output=1)
+    results = scipy.optimize.least_squares(fun=residuals, x0=initial_guess, args=(y_medians, slice_bounds))
```

Comment:
what is the difference between leastsq and least_squares? Is the former being deprecated?

Reply:
`leastsq` only calls the `lm` method for solving least squares, which is rarely used now. `least_squares` covers all methods, and defaults to `trp`

Reply:
also, the `xtol` of `least_squares` is more clearly defined depending on the method (easy to reproduce the calculation)

Reply:
in summary, we should only use `least_squares`, and call `lm` if we want that method in particular (no reason why though)